pkg/cache/v3/delta_test.go

```diff
@@ -175,24 +175,21 @@ func TestConcurrentSetDeltaWatch(t *testing.T) {
 			t.Run(fmt.Sprintf("worker%d", i), func(t *testing.T) {
 				t.Parallel()
 				id := fmt.Sprintf("%d", i%2)
-				var cancel func()
 				responses := make(chan cache.DeltaResponse, 1)
 				if i < 25 {
 					snap := cache.Snapshot{}
 					snap.Resources[types.Endpoint] = cache.NewResources(version, []types.Resource{resource.MakeEndpoint(clusterName, uint32(i))})
-					c.SetSnapshot(id, snap)
+					c.SetSnapshot(id, snap) // nolint:errcheck
 				} else {
-					if cancel != nil {
-						cancel()
-					}
-
-					cancel = c.CreateDeltaWatch(&discovery.DeltaDiscoveryRequest{
+					cancel := c.CreateDeltaWatch(&discovery.DeltaDiscoveryRequest{
 						Node: &core.Node{
 							Id: id,
 						},
 						TypeUrl:                rsrc.EndpointType,
 						ResourceNamesSubscribe: []string{clusterName},
 					}, stream.NewStreamState(false, make(map[string]string)), responses)
+
+					defer cancel()
```

Comment:
Did this not cause test failures?

Reply:
Nope, it was fine :)

Reply:
Nice, that was causing issues for me a while ago but I guess it's fixed